discovery/scaleway/baremetal.go

```diff
@@ -0,0 +1,143 @@
+// Copyright 2021 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package scaleway
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/prometheus/common/config"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/common/version"
+	"github.com/prometheus/prometheus/discovery/refresh"
+	"github.com/prometheus/prometheus/discovery/targetgroup"
+	"github.com/scaleway/scaleway-sdk-go/api/baremetal/v1"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+)
+
+type baremetalDiscovery struct {
+	*refresh.Discovery
+	Client    *scw.Client
+	Port      int
+	Zone      string
+	Project   string
+	AccessKey string
+	SecretKey string
+	Name      string
+	Tags      []string
+}
+
+const (
+	baremetalLabelPrefix = metaLabelPrefix + "baremetal_"
+
+	baremetalIDLabel        = baremetalLabelPrefix + "id"
+	baremetalZoneLabel      = baremetalLabelPrefix + "zone"
+	baremetalNameLabel      = baremetalLabelPrefix + "name"
+	baremetalStatusLabel    = baremetalLabelPrefix + "status"
+	baremetalTagsLabel      = baremetalLabelPrefix + "tags"
+	baremetalProjectIDLabel = baremetalLabelPrefix + "project_id"
+	baremetalIPLabel        = baremetalLabelPrefix + "ip"
+	baremetalIPVersionLabel = baremetalLabelPrefix + "ip_version"
+)
+
+func newBaremetalDiscovery(conf *SDConfig) (*baremetalDiscovery, error) {
+	d := &baremetalDiscovery{
+		Port:      conf.Port,
+		Zone:      conf.Zone,
+		Project:   conf.Project,
+		AccessKey: conf.AccessKey,
+		SecretKey: string(conf.SecretKey),
+		Name:      conf.FilterName,
+		Tags:      conf.FilterTags,
+	}
+
+	rt, err := config.NewRoundTripperFromConfig(conf.HTTPClientConfig, "scaleway_sd", false, false)
+	if err != nil {
+		return nil, err
+	}
+
+	profile, err := LoadProfile(conf)
+	if err != nil {
+		return nil, err
+	}
+	d.Client, err = scw.NewClient(
+		scw.WithHTTPClient(&http.Client{
+			Transport: rt,
+			Timeout:   time.Duration(conf.RefreshInterval),
+		}),
+		scw.WithUserAgent(fmt.Sprintf("Prometheus/%s", version.Version)),
```

Comment:
Isn't  a standard Prometheus user agent automatically set by the rt created by config.NewRoundTripperFromConfig(conf.HTTPClientConfig, "scaleway_sd", false, false) ?

Reply:
No, because the client is used for many other things. This is mainly used for tracking sessions with metrics, and select appropriate http settings.